src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -130,9 +134,16 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
                         .createSnapshotSync(cloud.getProjectId(), this.zone, this.getNodeName(), createSnapshotTimeout);
             }
 
-            // If the instance is running, attempt to terminate it. This is an async call and we
+            Map<String, String> filterLabel = ImmutableMap.of(CLOUD_ID_LABEL_KEY, cloud.getInstanceId());
+            var instanceExistsInCloud =
+                    cloud.getClient().listInstancesWithLabel(cloud.getProjectId(), filterLabel).stream()
+                            .anyMatch(instance -> instance.getName().equals(name));
+
+            // If the instance exists in the cloud, attempt to terminate it. This is an async call and we
             // return immediately, hoping for the best.
-            cloud.getClient().terminateInstanceAsync(cloud.getProjectId(), zone, name);
+            if (instanceExistsInCloud) {
```

Comment:
done into f6fc88ceb54d90dd77459fc4c5e20141aecc3858

Reply:
This is the actual fix for #471 and should not be removed.
**The Problem:**
When Jenkins tries to start a node and the quotas in the target cloud do not allow it, the instance is not started in the cloud but Jenkins thinks that a node was started. Now there is a Zombie node in Jenkins.

When trying to delete this Zombie node, Jenkins calls `cloud.getClient().terminateInstanceAsync` which throws an Exception since the Instance does not exist in the cloud. The Delete is aborted and the zombie node stays, forever undeletable.

**The Solution:**
I added a check if the node actually exists in the cloud before calling `cloud.getClient().terminateInstanceAsync`. If the node does not exist as an instance in the cloud, there is no need  to terminate it and the deletion can proceed.

**Why not just catch the exception thrown by `cloud.getClient().terminateInstanceAsync`?**
It is not clear from the exception why the terminate operation failed. If the instance exists in the cloud and the termination failed for some other reason, we do not want to delete the node in Jenkins.

Reply:
Ok, thanks for the detailed explanation.
I've reverted the changes.